src/main/scala/com/itechart/ny_accidents/service/WeatherMappingService.scala

```diff
@@ -1,52 +1,80 @@
 package com.itechart.ny_accidents.service
 
 import com.google.inject.{Inject, Singleton}
-import com.itechart.ny_accidents.constants.GeneralConstants
-import com.itechart.ny_accidents.constants.GeneralConstants.{FIRST_STATION_ID, LAST_STATION_ID}
+import com.itechart.ny_accidents.constants.GeneralConstants.{FIRST_STATION_ID, HASH_DIFFERENCE, LAST_STATION_ID}
 import com.itechart.ny_accidents.database.NYDataDatabase
 import com.itechart.ny_accidents.database.dao.WeatherDAO
-import com.itechart.ny_accidents.entity.{WeatherEntity, WeatherForAccident}
+import com.itechart.ny_accidents.entity.{WeatherEntity, WeatherForAccident, WeatherStation}
 import com.itechart.ny_accidents.parse.WeatherParser
+import com.itechart.ny_accidents.spark.Spark
+import com.itechart.ny_accidents.spark.Spark.sparkSql.implicits._
 import com.itechart.ny_accidents.utils.{DateUtils, PostgisUtils}
+import org.apache.spark.sql.Dataset
 
+import scala.annotation.tailrec
 import scala.concurrent.Await
 import scala.concurrent.duration.Duration
 import scala.util.Try
 
 @Singleton
 class WeatherMappingService @Inject()(weatherDAO: WeatherDAO, weatherParser: WeatherParser) {
 
-  private val allStations = Await.result(NYDataDatabase.database.run(weatherDAO.allStations()), Duration.Inf)
+
+  var counter = 1
+
+  private val allStations: Seq[WeatherStation] =
+    Await.result(NYDataDatabase.database.run(weatherDAO.allStations()), Duration.Inf)
   // map under have such structure -> Map[stationId, Map[TimeHash, Seq[WeatherEntity]]]
-  private val allWeather: Map[Int, Map[Long, Seq[WeatherEntity]]] = Await.result(
-    NYDataDatabase.database.run(weatherDAO.allWeather()), Duration.Inf)
-    .sortBy(weather => weather.dateTime)
-    .groupBy(weather => weather.stationId)
-    .mapValues(weathers => {
-      weathers.map(weather => (DateUtils.hashByDate(weather.dateTime), weather))
-        .groupBy(_._1)
-        .mapValues(seq => seq.map(_._2))
+  private val allWeather: Map[Int, Map[Long, Seq[WeatherEntity]]] =
+    Await.result(
+      NYDataDatabase.database.run(weatherDAO.allWeather()), Duration.Inf)
+      .sortBy(weather => weather.dateTime)
+      .groupBy(weather => weather.stationId)
+      .mapValues(weathers => {
+        weathers.map(weather => (DateUtils.hashByDate(weather.dateTime), weather))
+          .groupBy(_._1)
+          .mapValues(seq => seq.map(_._2))
+      })
+
+  private val newSparkWeather: Array[((Int, Long), WeatherForAccident)] = weatherDAO.loadAllWeatherSpark()
+    .map(row => ((row.getInt(1), DateUtils.hashByDate(row.getLong(2))), DateUtils.weatherForAccidentMapper(row)))
+    .collect()
+    .sortWith((a, b) => {
+      if (a._1._1 < b._1._1) {
```

Comment:
Don't understand what a and b means here

Reply:
https://github.com/itechartbigdatalab/NY-Accidents-Rate/pull/56/commits/76238650577cc2dfd2c8971fff3f0adb87a214b0